src/App.tsx

```diff
@@ -1,73 +1,123 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [filterValue, setFilterValue] = useState('');
+  const [debouncedFilterValue, setDebouncedFilterValue] = useState(filterValue);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const debouncedSetFilterValue = useCallback(
+    debounce((value: string) => {
+      setDebouncedFilterValue(value);
+    }, 300),
+    [],
+  );
+
+  const handleSelectedPerson = (person: Person | null) => {
+    setSelectedPerson(person);
+    setFilterValue(person ? person.name : '');
+    setIsDropdownOpen(false);
+  };
+
+  const handleChangeText = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setFilterValue(value);
+
+    debouncedSetFilterValue(value);
+
+    if (selectedPerson && value) {
+      setSelectedPerson(null);
+    }
+
+    setIsDropdownOpen(true);
+  };
+
+  const handleFocus = () => {
+    setIsDropdownOpen(true);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(user =>
+      user.name.toLowerCase().includes(debouncedFilterValue.toLowerCase()),
+    );
+  }, [debouncedFilterValue]);
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+        {selectedPerson && (
+          <h1 className="title" data-cy="title">
+            {`${selectedPerson?.name} (${selectedPerson?.born} - ${selectedPerson?.died})`}
+          </h1>
+        )}
+
+        {!selectedPerson && (
+          <h1 className="title" data-cy="title">
+            No selected person
+          </h1>
+        )}
```

Comment:
It will be shortest
```suggestion
        {selectedPerson ? (
          <h1 className="title" data-cy="title">
            {`${selectedPerson?.name} (${selectedPerson?.born} - ${selectedPerson?.died})`}
          </h1>
        ) : (
          <h1 className="title" data-cy="title">
            No selected person
          </h1>
        )}
```